index.js

```diff
@@ -114,7 +114,13 @@ function keysSorter(input) {
 }
 
 exports.extract = function (str) {
-	return str.split('?')[1] || '';
+	var queryStart = str.indexOf('?');
+	if (queryStart == -1) {
+		// no query string
+		return '';
+	} else {
+		return str.substring(queryStart+1)
+	}
```

Comment:
Even more logic would be needed if the library expects to work with URLs containing hashes (`#`).